kafka-client/src/test/java/dev/responsive/kafka/internal/db/mongo/MongoWindowTableTest.java

```diff
@@ -0,0 +1,129 @@
+import org.junit.jupiter.api.TestInfo;
+import org.junit.jupiter.api.extension.RegisterExtension;
+
+class MongoWindowTableTest {
+
+  @RegisterExtension
+  public static final ResponsiveExtension EXT = new ResponsiveExtension(StorageBackend.MONGO_DB);
+  private static final CollectionCreationOptions UNSHARDED = new CollectionCreationOptions(
+      false,
+      0
+  );
+  private static final byte[] DEFAULT_VALUE = new byte[] {1};
+
+  private String name;
+  private MongoClient client;
+
+  @BeforeEach
+  public void before(
+      final TestInfo info,
+      @ResponsiveConfigParam final Map<String, Object> props
+  ) {
+    name = info.getDisplayName().replace("()", "");
+
+    final String mongoConnection = (String) props.get(STORAGE_HOSTNAME_CONFIG);
+    client = SessionUtil.connect(mongoConnection, null, null);
+  }
+
+  @Test
+  public void shouldSucceedSimpleSetGet() {
+    // Given:
+    final WindowSegmentPartitioner partitioner = new WindowSegmentPartitioner(10_000L, 1_000L);
+    final var segment = partitioner.segmenter().activeSegments(0, 100).get(0);
+
+    final MongoWindowedTable table = new MongoWindowedTable(client, name, partitioner,
+        false, UNSHARDED
+    );
+    final var flushManager = table.init(0);
+    flushManager.updateOffsetAndStreamTime(0, 100);
+    flushManager.createSegment(segment);
+
+    // When:
+    final var byteKey = Bytes.wrap("key".getBytes());
+    var writer = flushManager.createWriter(segment);
+    writer.insert(
+        new WindowedKey(byteKey, 0),
+        DEFAULT_VALUE,
+        table.localEpoch(0)
+    );
+    writer.flush();
+
+    // Then:
+    var value = table.fetch(0, byteKey, 0);
+    assertThat(value, Matchers.equalTo(DEFAULT_VALUE));
+    value = table.fetch(0, byteKey, 100);
+    assertThat(value, Matchers.nullValue());
+    value = table.fetch(0, Bytes.wrap("other".getBytes()), 0);
+    assertThat(value, Matchers.nullValue());
+  }
+
+  @Test
+  public void shouldSucceedRangeSetGet() {
+    // Given:
+    final WindowSegmentPartitioner partitioner = new WindowSegmentPartitioner(10_000L, 1_000L);
+    final var segment = partitioner.segmenter().activeSegments(0, 100).get(0);
+
+    final MongoWindowedTable table = new MongoWindowedTable(client, name, partitioner,
+        false, UNSHARDED
+    );
+    final var flushManager = table.init(0);
+    flushManager.updateOffsetAndStreamTime(0, 100);
+    flushManager.createSegment(segment);
+
+    // When:
+    final var byteKey = Bytes.wrap("key".getBytes());
+    final var windowedKey = new WindowedKey(byteKey, 500);
+    var writer = flushManager.createWriter(segment);
+    writer.insert(
+        windowedKey,
+        DEFAULT_VALUE,
+        table.localEpoch(0)
+    );
+    writer.flush();
+
+    // Then:
+    var it = table.fetch(0, byteKey, 100, 600);
+    var kvs = new ArrayList<KeyValue<WindowedKey, byte[]>>();
+    it.forEachRemaining(kvs::add);
+
+    assertThat(kvs, Matchers.hasSize(1));
```

Comment:
nit: since we're trying to fetch a range, can we verify that multiple values can be returned instead of just one?

I know that's unrelated to this PR, but since this is somewhat of an integration test I think it's worth covering that part as well